config/src/main/java/com/typesafe/config/ConfigReferenceResolver.java

```diff
@@ -0,0 +1,23 @@
+package com.typesafe.config;
+
+/**
+ * Implement this interface to define how references that aren't defined within
+ * the config itself should be resolved.
+ */
+public interface ConfigReferenceResolver {
```

Comment:
I believe the docs mostly say "substitution" and not "reference" but I could misremember. `ConfigReference` is an internal class I think called that for some historical reason. `ConfigResolver` might be fine even, keep it shorter.